File: 3. Trapping Rainwater/BruteForceSol.js
/**
 * @param {number[]} height
 * @return {number}
 */
var trap = function (height) {
  let total = 0;

  for (let i = 0; i < height.length; i++) {
    const current = height[i];
    const maxL = maxBothSides(height, i, 1);
    const maxR = maxBothSides(height, i, 0);
    const smallerHeight = Math.min(maxL, maxR);
    const water = smallerHeight - current;

    if (water > 0) {
      total += water;
    }
  }

  return total;
};

//if mode = 1 left, else right
const maxBothSides = (arr, pointer, mode) => {
  let max = 0;
  if (mode) {
    for (let left = pointer - 1; left >= 0; left--) {
      max = Math.max(max, arr[left]);
    }
  } else {
    for (let right = pointer + 1; right < arr.length; right++) {
      max = Math.max(max, arr[right]);
    }
  }
  return max;
};

File: Container with Most Water/BruteForceSol.java
class Solution // Time O(n^2) Space O(1)
{
    public int maxArea(int[] height)
	{
        int maxArea = 0;
		
        for (int i = 0; i < height.length; i++)
		{
			for (int j = i + 1; j < height.length; j++)
			{
				int length = Math.min(height[i], height[j]);
				int width = j - i;
				int currArea =  length * width;
				maxArea = Math.max(maxArea, currArea);
			}
		}
        return maxArea;
    }
}